src/main.rs

```diff
@@ -83,13 +83,11 @@ fn main() -> Result<()> {
 
     let logger = slog::Logger::root(drain, slog::o!());
 
-    let listen_fds_found = listen_fds(true)?;
-    ensure!(
-        listen_fds_found == 1,
-        "expected one listen fd, got {}",
-        listen_fds_found
-    );
-    let listener = unsafe { UnixListener::from_raw_fd(LISTEN_FDS_START) };
+    let path = Path::new(SOCKET_PATH);
+    std::fs::create_dir_all(path.parent().expect("socket path has no parent"))?;
+    std::fs::remove_file(path).ok();
```

Comment:
isn't `let _ =` more idiomatic? (I am not actually sure)

Reply:
`let _ =` what?

Reply:
I think `std::fs::remove_file(path).ok()` is equivalent to saying "remove the file if possible, and I expect that if there are any errors with that we should proceed, including the one I expect most which is `ENOENT` which is fine, any other errors like permissions will probably surface when we try to bind".

Reply:
I'm used to seeing `let _ = std::fs::remove_file(path)`. But apparently silencing the warning with `.ok()` is also reasonable according to https://stackoverflow.com/questions/51141672/how-do-i-ignore-an-error-returned-from-a-rust-function-and-proceed-regardless